server/src/internalClusterTest/java/org/opensearch/index/shard/IndexShardIT.java

```diff
@@ -675,7 +675,7 @@ public final IndexShard newIndexShard(
             () -> {},
             RetentionLeaseSyncer.EMPTY,
             cbs,
-            new TransportCheckpointPublisher(client())
+            SegmentReplicationCheckpointPublisher.EMPTY
```

Comment:
Should we overload constructors here instead?  Perhaps something for when we subclass IndexShard for segrep.

Reply:
Yeah, given how intermingled things are right now, I'd rather leave this for when we have better separation of seg-rep and doc-rep in IndexShard